### PR TITLE
Add return type restrictions to crypto-related classes

### PR DESCRIPTION
**Pull request by oprypin**

### CODE REVIEW

**Review by straight-shoota — APPROVED** [review]

File: src/uuid.cr [hunk 1]
```diff
@@ -129,12 +129,12 @@ struct UUID
...
   end
 
-  def self.empty
+  def self.empty : self
     new(StaticArray(UInt8, 16).new(0_u8), UUID::Variant::NCS, UUID::Version::V4)
   end
 
   # Returns UUID variant.
-  def variant
+  def variant : UUID::Variant
```
Comment (Sija) [inline src/uuid.cr:137]: No need to use full paths.

```suggestion
  def variant : Variant
```

File: src/uuid.cr [hunk 2]
```diff
@@ -150,7 +150,7 @@ struct UUID
   end
 
   # Returns version based on RFC4122 format. See also `#variant`.
-  def version
+  def version : UUID::Version
```
Comment (Sija) [inline src/uuid.cr:153]:
```suggestion
  def version : Version
```

File: src/uuid.cr (hunk 1 above)
Comment (oprypin) [reply src/uuid.cr:137]: I think it's better as is. It's not returning a "Variant", it's returning a UUID of Variant kind

Comment (asterite) [reply src/uuid.cr:137]: That's a general naming problem we have. The type should have been called `VariantUUID`

**Pull request merged**